File: src/DataStructure/Array/2D/FindDiagonalOrder/javascript/findDiagonalOrder.test.js
import solution from './findDiagonalOrder.es3.cjs';
import runTestCase, { cases } from './testCases.js';

test.each(cases.map(c => [c.name, c]))(
  '[Data Structure/Array/2D/[498] Diagonal Traverse] - Case %#',
  (name, testCase) => expect(runTestCase(solution, testCase)).toEqual(testCase.expected),
);

File: src/DataStructure/Array/2D/FindDiagonalOrder/python/FindDiagonalOrder_test.py
import pytest
from FindDiagonalOrder.python.findDiagonalOrder import Solution
from FindDiagonalOrder.python.testCases import runTestCase, cases


@pytest.mark.parametrize('case', cases)
def test(case):
    solution = Solution()
    assert runTestCase(solution.findDiagonalOrder, case) == case['expected']
